File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    
    q1 = deque(queue1)
    q2 = deque(queue2)
    sum1 = sum(q1)
    sum2 = sum(q2)
    maxl = len(q1)
    
    if ((sum1 + sum2) % 2) != 0:
        return -1
    
    while q1 and q2:      
        if sum1 > sum2:
            a = q1.popleft() 
            q2.append(a)
            
            sum1 -= a
            sum2 += a
            answer += 1
            
        elif sum1 < sum2:
            a = q2.popleft() 
            q1.append(a)
            
            sum2 -= a
            sum1 += a
            answer += 1
            
        elif sum1 == sum2:
            return answer
        
        if answer > 300000:
            return -1
        
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        Queue<Integer> qu1 = new LinkedList<>();
        Queue<Integer> qu2 = new LinkedList<>();
        
        for (int i=0; i<queue1.length; i++) {
            qu1.offer(queue1[i]);
            qu2.offer(queue2[i]);
        }
        
        int count = 0;
        long sum1 = Arrays.stream(queue1).sum(), sum2 = Arrays.stream(queue2).sum();
        
        if ((sum1 + sum2) % 2 != 0) return -1; 
        
        while (!qu1.isEmpty() && !qu2.isEmpty()) {
            if (count > 300_000) return -1;
            
            if (sum1 > sum2) {
                int number = qu1.poll();
                qu2.offer(number);
                sum1 -= number;
                sum2 += number;
                count++;
            } else if (sum1 < sum2) {
                int number = qu2.poll();
                qu1.offer(number);
                sum2 -= number;
                sum1 += number;
                count++;
            } else {
                return count;
            }
        }
        
        return -1;
    }
}